lib/streamlit/elements/arrow_table.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018-2019 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import pyarrow as pa
+from streamlit import util
+from streamlit import type_util
+
+
+def marshall(proto, df, default_uuid):
+    if type_util.is_pandas_styler(df):
+        df = _marshall_styler(proto, df, default_uuid)
+
+    # Convert all dataframe values to strings as we
+    # only use them to show the result.
+    df = df.astype(str)
```

Comment:
Is the problem types? Can you add a note at why we need to do this?

Reply:
Will do!